### ISSUE TITLE
Following random user fails.

### ISSUE THREAD

**Issue by schiessle** [nextcloud/social#188]
I just tried to follow your @daita at c.nc.com. I clicked at your profile on the public time line and then the follow button which will show a spinner and nothing happens. The JS console shows:

````
TypeError: this.item is undefined[Learn More] social.js:250:174
	follow https://cloud.nextcloud.com/apps/social/js/social.js:250:174
	follow self-hosted:975:17
	e https://cloud.nextcloud.com/apps/social/js/social.js:7:12041
	Dr/o._withTask https://cloud.nextcloud.com/apps/social/js/social.js:7:44644

````

Btw, how do I follow random users, e.g. if I want to follow @bjoern@mastodon.social. Where can I enter the user?

**Comment by daita** [comment]
Bjorn: use to searchbar on the top right of the page.

**Comment by schiessle** [comment]
@daita searching for local users works great, but if I try to search for `@bjoern@mastodon.social` I see a spinner and nothing happens in the network console I see that the search request returned this error:

````
{"status":-1,"exception":"OCA\\Social\\Exceptions\\InvalidResourceException","message":"An exception occurred while executing 'INSERT INTO `oc_social_cache_actors` (`id`, `account`, `type`, `local`, `following`, `followers`, `inbox`, `shared_inbox`, `outbox`, `featured`, `url`, `preferred_username`, `name`, `summary`, `public_key`, `source`, `details`, `creation`, `icon_id`) VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [\"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\", \"bjoern@mastodon.social\", \"Person\", \"0\", \"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\/following\", \"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\/followers\", \"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\/inbox\", \"https:\\\/\\\/mastodon.social\\\/inbox\", \"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\/outbox\", \"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\/collections\\\/featured\", \"https:\\\/\\\/mastodon.social\\\/@bjoern\", \"bjoern\", \"Bj\\u00f6rn Schie\\u00dfle \\ud83c\\udf0d \\ud83c\\uddea\\ud83c\\uddfa\", \"Coordinator Germany &amp; GA member @fsfe; Co-founder of Nextcloud; Creating free, decentralized &amp; open cloud technology @nextcloud Opinions are my own\", \"-----BEGIN PUBLIC KEY-----\\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy5nNXNfkGUjxgXTqSFvA\\n+1X\\\/JafsYsZGpDOzrMnVi5yGbEpbh7T8RLP\\\/7dpWKwt4VaYmJKKF\\\/xX0zsHZP\\\/4+\\nocyBmMPTvZn3ep\\\/dPHBJWGnygOnaZPYvTCWEPTyP6tQsdmgku9qImQXD6wxrxp6g\\nULOWYx1VXiNRhEgotAqgyCAw4xIXYTo5egd2k\\\/3dk30EzjXGyDj+XKFeiA6V8cfn\\nxlSbJR9sJs\\\/HWVmghWnbYN9oy5I1KxCsH9C+YKGtO1YUHhWQNchZjQ832EPLja3r\\nVL9cQigg4mYo3cUjMn03bessnwrc6RH\\\/bmy\\\/4c7sufim04PRYhetvVV56OsLwkO5\\nvwIDAQAB\\n-----END PUBLIC KEY-----\\n\", \"{\\\"@context\\\":[\\\"https:\\\/\\\/www.w3.org\\\/ns\\\/activitystreams\\\",\\\"https:\\\/\\\/w3id.org\\\/security\\\/v1\\\",{\\\"manuallyApprovesFollowers\\\":\\\"as:manuallyApprovesFollowers\\\",\\\"sensitive\\\":\\\"as:sensitive\\\",\\\"movedTo\\\":{\\\"@id\\\":\\\"as:movedTo\\\",\\\"@type\\\":\\\"@id\\\"},\\\"Hashtag\\\":\\\"as:Hashtag\\\",\\\"ostatus\\\":\\\"http:\\\/\\\/ostatus.org#\\\",\\\"atomUri\\\":\\\"ostatus:atomUri\\\",\\\"inReplyToAtomUri\\\":\\\"ostatus:inReplyToAtomUri\\\",\\\"conversation\\\":\\\"ostatus:conversation\\\",\\\"toot\\\":\\\"http:\\\/\\\/joinmastodon.org\\\/ns#\\\",\\\"Emoji\\\":\\\"toot:Emoji\\\",\\\"focalPoint\\\":{\\\"@container\\\":\\\"@list\\\",\\\"@id\\\":\\\"toot:focalPoint\\\"},\\\"featured\\\":{\\\"@id\\\":\\\"toot:featured\\\",\\\"@type\\\":\\\"@id\\\"},\\\"schema\\\":\\\"http:\\\/\\\/schema.org#\\\",\\\"PropertyValue\\\":\\\"schema:PropertyValue\\\",\\\"value\\\":\\\"schema:value\\\"}],\\\"id\\\":\\\"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\",\\\"type\\\":\\\"Person\\\",\\\"following\\\":\\\"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\/following\\\",\\\"followers\\\":\\\"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\/followers\\\",\\\"inbox\\\":\\\"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\/inbox\\\",\\\"outbox\\\":\\\"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\/outbox\\\",\\\"featured\\\":\\\"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\/collections\\\/featured\\\",\\\"preferredUsername\\\":\\\"bjoern\\\",\\\"name\\\":\\\"Bj\\\\u00f6rn Schie\\\\u00dfle \\\\ud83c\\\\udf0d \\\\ud83c\\\\uddea\\\\ud83c\\\\uddfa\\\",\\\"summary\\\":\\\"\u003Cp\u003ECoordinator Germany &amp; GA member \u003Cspan class=\\\\\\\"h-card\\\\\\\"\u003E\u003Ca href=\\\\\\\"https:\\\/\\\/mastodon.social\\\/@fsfe\\\\\\\" class=\\\\\\\"u-url mention\\\\\\\"\u003E@\u003Cspan\u003Efsfe\u003C\\\/span\u003E\u003C\\\/a\u003E\u003C\\\/span\u003E; Co-founder of Nextcloud; Creating free, decentralized &amp; open cloud technology @nextcloud Opinions are my own\u003C\\\/p\u003E\\\",\\\"url\\\":\\\"https:\\\/\\\/mastodon.social\\\/@bjoern\\\",\\\"manuallyApprovesFollowers\\\":false,\\\"publicKey\\\":{\\\"id\\\":\\\"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern#main-key\\\",\\\"owner\\\":\\\"https:\\\/\\\/mastodon.social\\\/users\\\/bjoern\\\",\\\"publicKeyPem\\\":\\\"-----BEGIN PUBLIC KEY-----\\\\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy5nNXNfkGUjxgXTqSFvA\\\\n+1X\\\/JafsYsZGpDOzrMnVi5yGbEpbh7T8RLP\\\/7dpWKwt4VaYmJKKF\\\/xX0zsHZP\\\/4+\\\\nocyBmMPTvZn3ep\\\/dPHBJWGnygOnaZPYvTCWEPTyP6tQsdmgku9qImQXD6wxrxp6g\\\\nULOWYx1VXiNRhEgotAqgyCAw4xIXYTo5egd2k\\\/3dk30EzjXGyDj+XKFeiA6V8cfn\\\\nxlSbJR9sJs\\\/HWVmghWnbYN9oy5I1KxCsH9C+YKGtO1YUHhWQNchZjQ832EPLja3r\\\\nVL9cQigg4mYo3cUjMn03bessnwrc6RH\\\/bmy\\\/4c7sufim04PRYhetvVV56OsLwkO5\\\\nvwIDAQAB\\\\n-----END PUBLIC KEY-----\\\\n\\\"},\\\"tag\\\":[],\\\"attachment\\\":[{\\\"type\\\":\\\"PropertyValue\\\",\\\"name\\\":\\\"\\\\ud83d\\\\udccc Location\\\",\\\"value\\\":\\\"Stuttgart, Germany\\\"},{\\\"type\\\":\\\"PropertyValue\\\",\\\"name\\\":\\\"Pixelfed\\\",\\\"value\\\":\\\"\u003Ca href=\\\\\\\"https:\\\/\\\/pixelfed.social\\\/bjoern\\\\\\\" rel=\\\\\\\"me nofollow noopener\\\\\\\" target=\\\\\\\"_blank\\\\\\\"\u003E\u003Cspan class=\\\\\\\"invisible\\\\\\\"\u003Ehttps:\\\/\\\/\u003C\\\/span\u003E\u003Cspan class=\\\\\\\"\\\\\\\"\u003Epixelfed.social\\\/bjoern\u003C\\\/span\u003E\u003Cspan class=\\\\\\\"invisible\\\\\\\"\u003E\u003C\\\/span\u003E\u003C\\\/a\u003E\\\"},{\\\"type\\\":\\\"PropertyValue\\\",\\\"name\\\":\\\"\\\\ud83d\\\\udce8 Contact\\\",\\\"value\\\":\\\"\u003Ca href=\\\\\\\"https:\\\/\\\/www.schiessle.org\\\/contact\\\/\\\\\\\" rel=\\\\\\\"me nofollow noopener\\\\\\\" target=\\\\\\\"_blank\\\\\\\"\u003E\u003Cspan class=\\\\\\\"invisible\\\\\\\"\u003Ehttps:\\\/\\\/www.\u003C\\\/span\u003E\u003Cspan class=\\\\\\\"\\\\\\\"\u003Eschiessle.org\\\/contact\\\/\u003C\\\/span\u003E\u003Cspan class=\\\\\\\"invisible\\\\\\\"\u003E\u003C\\\/span\u003E\u003C\\\/a\u003E\\\"},{\\\"type\\\":\\\"PropertyValue\\\",\\\"name\\\":\\\"\\\\ud83c\\\\udf0d Homepage\\\",\\\"value\\\":\\\"\u003Ca href=\\\\\\\"https:\\\/\\\/www.schiessle.org\\\\\\\" rel=\\\\\\\"me nofollow noopener\\\\\\\" target=\\\\\\\"_blank\\\\\\\"\u003E\u003Cspan class=\\\\\\\"invisible\\\\\\\"\u003Ehttps:\\\/\\\/www.\u003C\\\/span\u003E\u003Cspan class=\\\\\\\"\\\\\\\"\u003Eschiessle.org\u003C\\\/span\u003E\u003Cspan class=\\\\\\\"invisible\\\\\\\"\u003E\u003C\\\/span\u003E\u003C\\\/a\u003E\\\"}],\\\"endpoints\\\":{\\\"sharedInbox\\\":\\\"https:\\\/\\\/mastodon.social\\\/inbox\\\"},\\\"icon\\\":{\\\"type\\\":\\\"Image\\\",\\\"mediaType\\\":\\\"image\\\/jpeg\\\",\\\"url\\\":\\\"https:\\\/\\\/files.mastodon.social\\\/accounts\\\/avatars\\\/000\\\/159\\\/379\\\/original\\\/ea2138d69d587746.jpg\\\"},\\\"image\\\":{\\\"type\\\":\\\"Image\\\",\\\"mediaType\\\":\\\"image\\\/jpeg\\\",\\\"url\\\":\\\"https:\\\/\\\/files.mastodon.social\\\/accounts\\\/headers\\\/000\\\/159\\\/379\\\/original\\\/50b13be83676236e.jpg\\\"},\\\"_address\\\":\\\"mastodon.social\\\",\\\"_path\\\":\\\"\\\/users\\\/bjoern\\\",\\\"_code\\\":200,\\\"_host\\\":\\\"mastodon.social\\\"}\", \"[]\", \"2018-12-09 18:00:24\", \"https:\\\/\\\/cloud.nextcloud.com\\\/documents\\\/g\\\/849e6229-4d29-416e-aada-16d97f467384\"]:\n\nSQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'source' at row 1"}
````

It seems like the search find my profile at mastodon.social but can't handle it.

**Comment by daita** [comment]
yes, it is a known issue that comes with using the searchbar from Nextcloud (#140)

When you're searching, we use the cache to get results. If you type the whole account, we can start checking if there is an instance behind the address, and then, if there is the account you're looking for on that instance.

Now, (I think) that because the search from the topright bar can be requested whenever the typing of the full address of the instance is not over, you might end up in a case where some request were trying to reach something that does not exists (remote instance). I think that if you type enter, sometime it works.

**Comment by daita** [comment]
Now there is also this issue when generating the cache with a too short field. I am working on this.

**Comment by daita** [comment]
#194 should do the trick

**Comment by juliushaertl** [comment]
Will be fixed with #217
